CommandScreen.c

```diff
@@ -49,21 +49,27 @@ static int CommandScreen_scanAscii(InfoScreen* this, const char* p, size_t total
 static int CommandScreen_scanWide(InfoScreen* this, const char* p, size_t total, char* line) {
    mbstate_t state;
    memset(&state, 0, sizeof(state));
-   int line_cols = 0;
-   int line_offset = 0, line_size = 0, last_spc_cols = -1, last_spc_offset = -1;
+   size_t bytes;
+   int line_cols = 0, line_offset = 0, line_size = 0, width = 0;
+   int last_spc_cols = -1, last_spc_offset = -1;
    for (size_t i = 0; i < total; ) {
       assert(line_offset >= 0 && (size_t)line_offset <= total);
-      wchar_t wc;
-      size_t bytes = mbrtowc(&wc, p + i, total - i, &state);
-      int width = wcwidth(wc);
-      if (bytes == (size_t)-1 || bytes == (size_t)-2 || bytes == 0 || width < 0) {
+      unsigned char c = (unsigned char)p[i];
+      if (c < 0x80) { // skip mbrtowc for ASCII characters
```

Comment:
This lacks checking or assertion on control characters.

Reply:
Code point range [0x00, 0x1F] and 0x7F are unprintable. You shouldn't assume the width would be 1 here.

Reply:
Got you, appreciate your feedback, but again, for now I’d prefer to focus on resolving the line wrapping issue first.

Reply:
I think your effort on trying to wrap the commnd line on space boundaries is unjustified. Unicode line breaking is more complicated than that, and in case of file paths, it might be more appropriate to wrap at path separators (that is, slashes) rather than spaces.

And since you are working with Unicode characters with variable display width, the issue #1603 would be relevant. I think you can read that first.